### ISSUE TITLE
Share link with preset expiration/burn configuration

### ISSUE THREAD

**Issue by dillonstreator** [PrivateBin/PrivateBin#1453]
## The problem

I'd like to share privatebin.net with someone and have the expiration and burn after reading preset to a desired value and populated in the interface when the user navigates to the home page.

## The solution

Make the client application aware of two query parameters `burn` and `exp` to initialize the form inputs

https://privatebin.net?burn=true&exp=30m
Defaults burn after reading checkbox to true and expiration to 30 minutes

https://privatebin.net?burn=false&exp=2wk
Defaults burn after reading checkbox to false (no-op given this is already the default) and expiration to 2 weeks

Don't feel strongly about the naming convention for expiration duration parsing but here a couple libraries that could be used
https://www.npmjs.com/package/parse-duration
https://www.npmjs.com/package/duration

If a `burn` or `exp` value is provided that cannot be parsed, it should just be ignored
A new button could be added to copy the current configuration to clipboard

**Comment by elrido** [comment]
It is already possible to use PrivateBin with your own choice of defaults for burn-after-reading and expiration preselected, by way of running your own instance. privatebin.net is just this projects demo-instance, but if you run your own instance, you can configure these defaults as you like, in your instances conf.php file:

1. There is an [option `burnafterreadingselected`](https://github.com/PrivateBin/PrivateBin/wiki/Configuration#burnafterreadingselected) that lets you pre-select that checkbox for newly created pastes.
2. There are both [options to configure your own list of expiration options, as well as to select your desired default choice](https://github.com/PrivateBin/PrivateBin/wiki/Configuration#expiration):

https://github.com/PrivateBin/PrivateBin/blob/3460f20e40241f5c5ac54de3343710166bc6ceb8/cfg/conf.sample.php#L127-L143

As can be seen in the above sample configuration, we already have a format for these durations, which both the client- and server-side logic can parse, to format them for human consumption:
https://github.com/PrivateBin/PrivateBin/blob/3460f20e40241f5c5ac54de3343710166bc6ceb8/js/privatebin.js#L358-L363

The configuration file is responsible to turn them into seconds - no need to parse the string and recalculate that, over and over again.

**Comment by dillonstreator** [comment]
@elrido appreciate the pointers and references. As I understand it, this doesn't directly support the use case I'm requesting. My request would allow for a user to dynamically change the default configuration via the URL. I may want to configure burn after reading as enabled in my conf.php but then share a link to my privatebin instance with burn after reading turned off for a particular scenario. With this change, as a user, I can override the configured defaults on an as needed basis.

**Comment by michealespinola** [comment]
I agree that this would make the service more versatile to be able to essentially (pre) share the link being [suggested] to a new user to use, with appropriate default settings [configured]. Being a web form, this is relatively cooked-in functionality.

This would help reduce new-user mistakes.

edit: edits in [brackets]

**Comment by dillonstreator** [comment]
@michealespinola exactly - this is what I ended up doing in my privatebin alternative application ([crypt.fyi](https://www.crypt.fyi)) which has dynamic URL pre-configuration. e.g. https://www.crypt.fyi/new?ttl=3d&b=false&rc=5&fc=5

**Comment by rugk** [comment]
It's essentially a Javascript adjustment to read query parameters and then set them as a preset. I think this is a valid feature request. 

Implementation note: use the URL interface https://developer.mozilla.org/en-US/docs/Web/API/URL for properly parsing the URL please, not wild RegEx or so. 

@dillonstreator if you already implemented it, it would be great if you could submit a PR or so.

**Comment by rugk** [comment]
Also implemention note: maybe the server admin should have an opt-out to disable "dynamic configuration by url" or how we'd call this feature? As maybe an admin does not want that for some reason? (imho as said it can be enabled by default though)

**Comment by michealespinola** [comment]
An opt-out is a great idea as well, if the admin would like their own defaults to always be enforced
